server/sonar-server/src/main/java/org/sonar/server/permission/PermissionService.java

```diff
@@ -76,6 +77,11 @@ public PermissionService(DbClient dbClient, PermissionRepository permissionRepos
     return GlobalPermissions.ALL;
   }
 
+  public UserWithPermissionQueryResult findUsersWithPermission(PermissionQuery query) {
```

Comment:
I don't think PermissionService must be removed, it must become the main entry point for anything related to permission. There's a lot of logic to add/delete permissions. PermissionFinder could possibly be merged in PermissionService, not the other way around. For the moment, it can stay as is.